File: app/src/main/java/de/uni_marburg/mathematik/ds/serval/model/comparators/TimeComparator.java
package de.uni_marburg.mathematik.ds.serval.model.comparators;

import java.util.Comparator;

import de.uni_marburg.mathematik.ds.serval.model.event.Event;

/**
 * /**
 * Compares {@link Event events} based on their {@link Event#time time}.
 *
 * @param <T> Type of the {@link Event event}
 */
public class TimeComparator<T extends Event> implements Comparator<T> {
    
    @Override
    public int compare(T t, T t1) {
        return t.getTime().compareTo(t1.getTime());
    }
}

File: app/src/main/java/de/uni_marburg/mathematik/ds/serval/model/comparators/TimeComparator.kt
package de.uni_marburg.mathematik.ds.serval.model.comparators

import de.uni_marburg.mathematik.ds.serval.model.event.Event
import java.util.*

/** Compares [events][Event] based on their occurrence time. **/
class TimeComparator : Comparator<Event> {

    override fun compare(event1: Event, event2: Event): Int =
            compareValues(event1.time, event2.time)

}
